docs/next/components/mdx/MDXComponents.tsx

```diff
@@ -535,6 +576,38 @@ function classNames(...classes) {
 }
 
 const TabGroup: React.FC<{children: any; persistentKey?: string}> = ({children, persistentKey}) => {
+  const [selectedTab, setSelectedTab] = useState(0);
+  const [anchor, _setAnchor] = useHash();
+
+  const [anchorsInChildren, setAnchorsInChildren] = useState<{[anchor: string]: number}>({});
+  const handleTabs = useCallback((node: HTMLElement) => {
+    const out = {};
+
+    // Once the tabs render, get the list of element IDs and the map to the
+    // tab index they are in
+    for (let i = 0; i < node.children.length; i++) {
+      const tab = node.children[i] as HTMLElement;
+      for (const element of tab.querySelectorAll('[id]')) {
+        out[element.id] = i;
+      }
+    }
+    setAnchorsInChildren(out);
+  }, []);
+
+  useEffect(() => {
+    const anchorWithoutHash = anchor.substring(1);
+    if (anchorWithoutHash in anchorsInChildren) {
+      const tabIdx = anchorsInChildren[anchorWithoutHash];
+
+      // Scroll page to the hash after re-render
+      setSelectedTab(tabIdx);
+      setTimeout(() => {
+        const elem = document.getElementById(anchorWithoutHash);
+        elem.scrollIntoView();
```

Comment:
Does the docs build run a TS check? This looks like it might need to be `elem?.scrollIntoView()`, since `getElementById` might return null.